src/mc_tools.jl

```diff
@@ -295,8 +295,115 @@ function mc_sample_path!(mc::MarkovChain, samples::Array)
     Void
 end
 
+"""
+Simulate time series of state transitions of the Markov chain `mc`.
+
+##### Arguments
+
+- `mc::MarkovChain` : MarkovChain instance.
+- `ts_length::Int` : Length of each simulation.
+- `init::Vector{Int}` : Vector containing initial states.
+- `;num_reps::Union{Int, Void}(nothing)` : Number of repetitions of simulation.
+
+##### Returns
+
+- `X::Array{Int}` : Array containing the sample paths as columns, of shape
+(ts_length, k), where k = length(init) if num_reps=nothing, and k =
+length(init)*num_reps otherwise.
+
+"""
+function simulate(mc::MarkovChain,
+                  ts_length::Int,
+                  init::Vector{Int};
+                  num_reps::Union{Int, Void}=nothing)
+    k = length(init)
+    if num_reps == nothing
+        num_reps = 1
+    end
+    k *= num_reps
+    X = Array(Int, ts_length, k)
+    X[1, :] = repmat(init, num_reps)
+
+    simulate!(mc, X)
+    return X
+end
+
+"""
+Simulate time series of state transitions of the Markov chain `mc`.
+
+##### Arguments
+
+- `mc::MarkovChain` : MarkovChain instance.
+- `ts_length::Int` : Length of each simulation.
+- `init::Int` : Initial state.
+- `;num_reps::Union{Int, Void}(nothing)` : Number of repetitions of simulation.
+
+##### Returns
+
+- `X::Array{Int}` : Array containing the sample path(s) as column(s), of shape
+(ts_length,) if num_reps=nothing; of shape (ts_length, num_reps) otherwise.
+
+"""
+function simulate(mc::MarkovChain,
+                  ts_length::Int,
+                  init::Int;
+                  num_reps::Union{Int, Void}=nothing)
```

Comment:
@spencerlyon2 Thank you for the comments.

Here I wanted to follow the convention in the python version, where the returned array is 1-dimensional when `init` is a scalar and `num_reps` is not specified, while it is 2-dimensional when `num_reps` is specified (even if `num_reps = 1`).